docs/psql.md

```diff
@@ -0,0 +1,52 @@
+# Google Cloud Spanner PostgreSQL Adapter - PSQL Support
+
+PGAdapter supports [psql](https://www.postgresql.org/docs/current/app-psql.html) versions 11, 12, 13 and 14.
+
+## Usage
+
+First start PGAdapter in `psql` mode (see [README](../README.md) for more possibilities on how to run
+PGAdapter).
+
+```shell
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
+docker pull us-west1-docker.pkg.dev/cloud-spanner-pg-adapter/pgadapter-docker-images/pgadapter
+docker run \
+  -d -p 5432:5432 \
+  -v ${GOOGLE_APPLICATION_CREDENTIALS}:${GOOGLE_APPLICATION_CREDENTIALS}:ro \
+  -e GOOGLE_APPLICATION_CREDENTIALS \
+  us-west1-docker.pkg.dev/cloud-spanner-pg-adapter/pgadapter-docker-images/pgadapter \
+  -p my-project -i my-instance -d my-database \
+  -x -q
+```
+
+Note:
+- The `-x` option tells PGAdapter to accept incoming connections from other hosts than localhost.
+  This is necessary when running PGAdapter in a Docker container, as you will be connecting to it
+  from outside the container, which will be seen as a connection from a different host.
+- The `-q` option tells PGAdapter to translate queries from `psql` for meta commands like `\dt` into
+  queries that are supported by Cloud Spanner.
+
+Then connect to PGAdapter using `psql`:
+
+```shell
+psql -h localhost
+```
+
+## Running alongside a local PostgreSQL server
+
+If you have PostgreSQL installed locally on your machine, it is highly likely that PostgreSQL is
+already using port 5432. You then need to run PGAdapter on a different port, and specify this port
+when starting `psql` to connect to PGAdapter instead of the local PostgreSQL server:
+
+```shell
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
+docker pull us-west1-docker.pkg.dev/cloud-spanner-pg-adapter/pgadapter-docker-images/pgadapter
+docker run \
+  -d -p 5433:5432 \
+  -v ${GOOGLE_APPLICATION_CREDENTIALS}:${GOOGLE_APPLICATION_CREDENTIALS}:ro \
+  -e GOOGLE_APPLICATION_CREDENTIALS \
+  us-west1-docker.pkg.dev/cloud-spanner-pg-adapter/pgadapter-docker-images/pgadapter \
+  -p my-project -i my-instance -d my-database \
+  -x -q
```

Comment:
can u highlight -q 

all other options are common to all tools it seems and therefore should be de-duped

Reply:
I've reorganized the jdbc, pgx and psql documents a little to try to highlight the specific options that are needed for the specific tool. I don't think it's good to deduplicate the examples by removing the parameters that are common, as people generally want commands that they can copy-paste directly.